public/resources/core/js/services/party.individual.service.js

```diff
@@ -138,93 +143,125 @@
             create: create,
             detail: detail,
             update: update,
-            launch: launch
+            launch: launch,
+	    getCurrentOrg: getCurrentOrg,
+	    hasAdminRole: hasAdminRole
         };
 
-        function create(data) {
+	function getCurrentOrg() {
+	    var org = User.loggedUser.currentUser;
+	    return User.loggedUser.organizations.find( x => x.id === org.id);
+	};
+	
+	function hasAdminRole() {
+	    var org = User.loggedUser.organizations.find(
+		x => x.id === User.loggedUser.currentUser.id);
+	    return org.roles.findIndex(x => x.name === "Admin") > -1;
+	};
+
+	function process(func, params) {
+
+	    var deferred = params.pop();
+	    var resol = function (partyObj) {
+		deferred.resolve(partyObj);
+	    };
+
+	    var rejec = function (response) {
+		deferred.reject(response);
+	    };
+	    params.push(resol, rejec);
+	    
+	    func.apply(null, params);
+
+	};
+
+        function create(data, isOrg) {
             var deferred = $q.defer();
 
-            Individual.save(data, function (individualCreated) {
-                deferred.resolve(individualCreated);
-            }, function (response) {
-                deferred.reject(response);
-            });
+	    if(!isOrg){
+		process(Individual.save, [data, deferred]);
+	    } else {
+		process(Organization.save, [data, deferred]);
+	    }
 
             return deferred.promise;
-        }
+        };
 
-        function detail(id) {
+        function detail(id, isOrg) {
             var deferred = $q.defer();
             var params = {
                 id: id
             };
+	    if(!isOrg){
+		process(Individual.get, [params, deferred]);
```

Comment:
This is not equivalent, note that in the old code extendContactMedium(individual); was called before the resolve